pkg/state/state.go

```diff
@@ -2976,18 +2976,27 @@ func (st *HelmState) namespaceAndValuesFlags(helm helmexec.Interface, release *R
 	return flags, files, nil
 }
 
-func (st *HelmState) setFlags(setValues []SetValue) ([]string, error) {
+func (st *HelmState) setFlags(release *ReleaseSpec) ([]string, error) {
 	var flags []string
 
-	for _, set := range setValues {
+	for _, set := range release.SetValues {
 		if set.Value != "" {
 			renderedValue, err := renderValsSecrets(st.valsRuntime, set.Value)
 			if err != nil {
 				return nil, err
 			}
 			flags = append(flags, "--set", fmt.Sprintf("%s=%s", escape(set.Name), escape(renderedValue[0])))
 		} else if set.File != "" {
-			flags = append(flags, "--set-file", fmt.Sprintf("%s=%s", escape(set.Name), st.storage().normalizePath(set.File)))
+			outputValuesFile, err := st.GenerateOutputFilePath(release, set.File)
```

Comment:
I realized that this might be wrong as the GenerateOutputFilePath is dedicated to generating a dynamic path for the `helmfile write-values` command.
I think this logic should be migrated to the existing ExecuteTemplateExpressions found at https://github.com/helmfile/helmfile/blob/44c26e6639989b7a55ff4d85e52aec280ec477c1/pkg/state/release.go#L12 for consistency.